sa/saro.go

```diff
@@ -1302,16 +1302,22 @@ func (ssa *SQLStorageAuthority) GetRevokedCerts(req *sapb.GetRevokedCertsRequest
 // table to implement GetRevokedCerts. It must only be called when the request
 // contains a non-zero ShardIdx.
 func (ssa *SQLStorageAuthorityRO) getRevokedCertsFromRevokedCertificatesTable(req *sapb.GetRevokedCertsRequest, stream sapb.StorageAuthorityReadOnly_GetRevokedCertsServer) error {
+	if req.ShardIdx == 0 {
+		return errors.New("can't select shard 0 from revokedCertificates table")
+	}
+
 	atTime := time.Unix(0, req.RevokedBeforeNS)
 
 	clauses := `
 		WHERE issuerID = ?
 		AND shardIdx = ?
-		AND notAfter >= ?`
+		AND notAfterHour >= ?`
```

Comment:
I think technically this can become a `>` now. Imagine certificate XYZ, with a `NotAfter` of 13:00. It would be stored in `revokecCertificates` with a `notAfterHour` of 14:00. So if `ExpiresAfterNS` is 13:00, `notAfterHour > 13:00` would include certificate XYZ.

Perhaps it's just worth being over-inclusive here in case we got the math wrong, but I wanted to write it down to make sure I understand correctly.

Reply:
Yep, I had the same thought and agree with your math, but chose to leave it >= as a "more obviously correct" measure.